internal/cron/verify_domain_ownership.go

```diff
@@ -0,0 +1,88 @@
+package cron
+
+import (
+	"context"
+	"errors"
+	"net"
+
+	"github.com/oursky/pageship/internal/db"
+	"github.com/oursky/pageship/internal/models"
+	"github.com/oursky/pageship/internal/time"
+	"go.uber.org/zap"
+)
+
+type VerifyDomainOwnership struct {
+	Schedule                     string
+	DB                           db.DB
+	MaxConsumeActiveDomainCount  uint
+	MaxConsumePendingDomainCount uint
+}
+
+func (v *VerifyDomainOwnership) Name() string { return "verify-domain-ownership" }
+
+func (v *VerifyDomainOwnership) CronSchedule() string { return v.Schedule }
+
+func (v *VerifyDomainOwnership) Run(ctx context.Context, logger *zap.Logger) error {
+	now := time.SystemClock.Now().UTC()
+
+	return db.WithTx(ctx, v.DB, func(c db.Tx) error {
+		verified := true
+		unverified := false
+		domainVerifications := []*models.DomainVerification{}
+		verifiedRecords, _ := c.ListDomainVerifications(ctx, nil, &v.MaxConsumeActiveDomainCount, &verified)
+		unverifedRecords, _ := c.ListDomainVerifications(ctx, nil, &v.MaxConsumePendingDomainCount, &unverified)
+		for _, verifiedRecord := range verifiedRecords {
+			domainVerifications = append(domainVerifications, verifiedRecord)
+		}
+		for _, unverifiedRecord := range unverifedRecords {
+			domainVerifications = append(domainVerifications, unverifiedRecord)
+		}
+
+		var consumedCount = 0
+		for _, domainVerification := range domainVerifications {
+			key, value := domainVerification.GetTxtRecord()
+			values, err := net.LookupTXT(key)
+			var dnsErr *net.DNSError
+			if err != nil && errors.As(err, &dnsErr) && !dnsErr.IsNotFound {
+				continue
+			}
+			consumedCount += 1
+			matched := false
+			for _, v := range values {
+				if v == value {
+					matched = true
+					break
+				}
+			}
+			domain, err := c.GetDomainByName(ctx, domainVerification.Domain)
```

Comment:
The error is not checked correctly: if `matched` is true and `err` is non-nil (and not `ErrDomainNotFound`), it is ignored.

Please move the error checking right after the operation.